PSReadLine/ReadLine.vi.cs

```diff
@@ -220,16 +220,46 @@ public static void DeleteToEnd(ConsoleKeyInfo? key = null, object arg = null)
                 return;
             }
 
-            _clipboard.Record(_singleton._buffer, _singleton._current, _singleton._buffer.Length - _singleton._current);
-            _singleton.SaveEditItem(EditItemDelete.Create(
-                _clipboard,
-                _singleton._current,
-                DeleteToEnd,
-                arg
-                ));
-            _singleton._buffer.Remove(_singleton._current, _singleton._buffer.Length - _singleton._current);
-            _singleton._current = Math.Max(0, _singleton._buffer.Length - 1);
-            _singleton.Render();
+            var lineCount = _singleton.GetLogicalLineCount();
+            var lineIndex = _singleton.GetLogicalLineNumber() - 1;
+
+            if (TryGetArgAsInt(arg, out var requestedLineCount, 1))
+            {
+                var targetLineIndex = lineIndex + requestedLineCount - 1;
+                if (targetLineIndex >= _singleton.GetLogicalLineCount())
+                {
+                    targetLineIndex = _singleton.GetLogicalLineCount() - 1;
+                }

```

Comment:
How about replacing both calls to `_singleton.GetLogicalLineCount()` here with `lineCount`?